src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateClonedParameterSymbolForBeginAndEndInvoke.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SourceDelegateClonedParameterSymbolForBeginAndEndInvoke : SourceClonedParameterSymbol
+    {
+        internal SourceDelegateClonedParameterSymbolForBeginAndEndInvoke(SourceParameterSymbol originalParam, SourceDelegateMethodSymbol newOwner, int newOrdinal)
+            : base(originalParam, newOwner, newOrdinal, suppressOptional: true)
+        {
+        }
+
+        internal override bool IsCallerFilePath => _originalParam.IsCallerFilePath;
+
+        internal override bool IsCallerLineNumber => _originalParam.IsCallerLineNumber;
+
+        internal override bool IsCallerMemberName => _originalParam.IsCallerMemberName;
+
+        // From code review: Could we just return -1 for now and see if anyone would ask us to support the scenarios.
+        internal override int CallerArgumentExpressionParameterIndex => -1;
```

Comment:
>-1 [](http://example.com/codeflow?start=72&length=2)

I'm unclear why this would be ok to return -1 from and the `IsCallerX` members shouldn't just return false. If we think think one of these is hittable, shouldn't they all be?

Reply:
@333fred They are all hittable. But @AlekseyTs has requested to drop support for begin and end invoke.

https://github.com/dotnet/roslyn/pull/54355#discussion_r659891931

Reply:
I see, thanks for the comment. In that case, we should update the comment to be an explicit statement of that info, rather than a question.